static/js/util/test_utils.js

```diff
@@ -159,6 +160,74 @@ export const localStorageMock = (init: any = {}) => {
   };
 };
 
+export class GoogleMapsStub {
+  sandbox: Sandbox;
+  getPlacePredictions: any;
+  geocode: any;
+
+  autocompleteSuggestions = [{
+    "description": "Paris, France",
+    "id": "691b237b0322f28988f3ce03e321ff72a12167fd",
+    "matched_substrings": [{"length": 5, "offset": 0}],
+    "place_id": "ChIJD7fiBh9u5kcRYJSMaMOCCwQ",
+    "reference": "CjQlAAAA_KB6EEceSTfkteSSF6U0",
+    "terms": [
+      {offset: 0, value: "Paris"},
+      {offset: 7, value: "France"},
+    ],
+    "types": ["locality", "political", "geocode"]
+  }];
+
+  geocodeResults = [{
+    address_components: [
+      {long_name: "123 Main Street", short_name: "123 Main St", types: ["street_address"]},
+      {long_name: "Anytown", short_name: "Anytown", types: ["locality"]},
+      {long_name: "RandoState", short_name: "RS", types: ["administrative_area_level_1"]},
+      {long_name: "United States", short_name: "USA", types: ["country"]},
+    ],
+    geometry: {
+      location: {
+        lat: () => 48.859,
+        lng: () => 2.207,
+      }
+    }
+  }];
+
+  constructor() {
+    this.sandbox = sinon.sandbox.create();
+    this.getPlacePredictions = this.sandbox.stub().callsArgWith(1, this.autocompleteSuggestions);
+    this.geocode = this.sandbox.stub().callsArgWith(1, this.geocodeResults, 'OK');
+
+    const that = this;
+    class FakeAutocompleteService {
+      getPlacePredictions = that.getPlacePredictions;
+    }
+    class FakeGeocoder {
+      geocode = that.geocode;
+    }
+
+    let google = {
+      maps: {
+        Geocoder: FakeGeocoder,
+        GeocoderStatus: {
+          OK: 'OK'
+        },
+        places: {
+          AutocompleteService: FakeAutocompleteService
+        }
+      }
+    };
+
+    window.google = google;
+  }
+
+  cleanup() {
+    this.sandbox.restore();
+    delete window.google;
+  }
+}
```

Comment:
My concern is that most of our pages do _not_ load the Google Maps API, but it's a common-enough library that other libraries may detect it and work differently depending on if the library is loaded or not. I think it's worthwhile to test _without_ a `google` variable in the global context, to more closely simulate how most of our pages will act.

That being said, this is a very minor concern, and I rather doubt that any of the libraries we're using will detect the presence of the `google` variable. I'm willing to be overruled on this if you think that the reduced boilerplate code is worth the imperfect testing environment.